casbin/core_enforcer.py

```diff
@@ -177,16 +177,34 @@ def init_rm_map(self):
 
     def load_policy(self):
         """reloads the policy from file/database."""
+        need_to_rebuild = False
+        new_model = copy.copy(self.model)
+        new_model.clear_policy()
 
-        self.model.clear_policy()
-        self.adapter.load_policy(self.model)
+        try:
 
-        self.model.sort_policies_by_priority()
+            self.adapter.load_policy(new_model)
 
-        self.init_rm_map()
-        self.model.print_policy()
-        if self.auto_build_role_links:
-            self.build_role_links()
+            new_model.sort_policies_by_priority()
+
+            self.init_rm_map()
+            self.model.print_policy()
+
+            if self.auto_build_role_links:
+
+                need_to_rebuild = True
+                for rm in self.rm_map.values():
+                    rm.clear()
+
+                new_model.build_role_links(self.rm_map)
+                self.build_role_links()
+
+            self.model = new_model
+
+        except:
+
+            if self.auto_build_role_links and need_to_rebuild:
+                self.build_role_links()
```

Comment:
@ffyuanda  Here to throw the caught exception again.

Reply:
@techoner Just update, I catch the error and throw it after everything else.